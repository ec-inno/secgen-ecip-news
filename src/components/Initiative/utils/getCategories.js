const getCategories = t => {
  return [
    { value: 'any', label: t('All categories') },
    { value: 'AGRI', label: t('Agriculture') },
    { value: 'DEVCO', label: t('Aid and development cooperation') },
    { value: 'EURO', label: t('Business and economy') },
    { value: 'SANTE', label: t('Consumers and health') },
    { value: 'CULT', label: t('Culture and media') },
    { value: 'DECO', label: t('Digital economy and society') },
    { value: 'EDU', label: t('Education, youth and sport') },
    { value: 'EMPL', label: t('Employment and social affairs') },
    { value: 'ENER', label: t('Energy') },
    { value: 'ENV', label: t('Environment and climate') },
    { value: 'TRADE', label: t('External trade and relations') },
    { value: 'JUST', label: t('Justice and fundamental rights') },
    { value: 'MARE', label: t('Maritime affairs and fisheries') },
    { value: 'MIGR', label: t('Migration and asylum') },
    { value: 'REGIO', label: t('Regional development') },
    { value: 'RSH', label: t('Research') },
    { value: 'SEC', label: t('Security') },
    { value: 'TRA', label: t('Transport') },
    { value: 'REGFRA', label: t('REGFRA') },
    { value: 'REGECI', label: t('REGECI') },
    { value: 'COLLECI', label: t('COLLECI') },
    { value: 'EXAECI', label: t('EXAECI') },
  ];
};

export default getCategories;
