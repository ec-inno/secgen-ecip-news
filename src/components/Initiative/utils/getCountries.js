const getCountriesTranslated = t => {
  return [
    { value: 'bg', label: t('Bulgaria') },
    { value: 'es', label: t('Spain') },
    { value: 'cs', label: t('Czech Republic') },
    { value: 'da', label: t('Denmark') },
    { value: 'de', label: t('Germany') },
    { value: 'et', label: t('Estonia') },
    { value: 'el', label: t('Greece') },
    { value: 'fr', label: t('France') },
    { value: 'ga', label: t('Ireland') },
    { value: 'hr', label: t('Croatia') },
    { value: 'it', label: t('Italia') },
    { value: 'lv', label: t('Latvia') },
    { value: 'lt', label: t('Lithuania') },
    { value: 'hu', label: t('Hungary') },
    { value: 'mt', label: t('Malta') },
    { value: 'nl', label: t('Netherlands') },
    { value: 'pl', label: t('Poland') },
    { value: 'pt', label: t('Portugal') },
    { value: 'ro', label: t('Romania') },
    { value: 'sk', label: t('Slovakia') },
    { value: 'sl', label: t('Slovenia') },
    { value: 'fi', label: t('Finland') },
    { value: 'sv', label: t('Sweden') },
  ];
};

export default getCountriesTranslated;
