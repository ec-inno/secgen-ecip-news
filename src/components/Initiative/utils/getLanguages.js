const getLanguagesTranslated = t => {
  return [
    { value: 'bg', label: t('Bulgarian') },
    { value: 'es', label: t('Spanish') },
    { value: 'cs', label: t('Czech') },
    { value: 'da', label: t('Dansk') },
    { value: 'de', label: t('German') },
    { value: 'et', label: t('Estonian') },
    { value: 'el', label: t('Greek') },
    { value: 'en', label: t('English') },
    { value: 'fr', label: t('French') },
    { value: 'ga', label: t('Irish') },
    { value: 'hr', label: t('Croatian') },
    { value: 'it', label: t('Italian') },
    { value: 'lv', label: t('Latvian') },
    { value: 'lt', label: t('Lithuanian') },
    { value: 'hu', label: t('Hungarian') },
    { value: 'mt', label: t('Maltese') },
    { value: 'nl', label: t('Dutch') },
    { value: 'pl', label: t('Polish') },
    { value: 'pt', label: t('Portuguese') },
    { value: 'ro', label: t('Romanian') },
    { value: 'sk', label: t('Slovak') },
    { value: 'sl', label: t('Slovene') },
    { value: 'fi', label: t('Finnish') },
    { value: 'sv', label: t('Swedish') },
  ];
};

export default getLanguagesTranslated;
