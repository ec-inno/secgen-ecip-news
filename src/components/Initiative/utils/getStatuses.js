// These are values for filters, not resource path!
// Resource path values are LATEST, ONGOING, ANSWERED, ALL, REFUSED
const getStatuses = t => {
  return [
    { value: 'ALL', label: t('All') },
    { value: 'OPEN', label: t('Open') },
    { value: 'SUCCESSFUL', label: t('Successful') },
    { value: 'ARCHIVED', label: t('Archived') },
  ];
};

export default getStatuses;
