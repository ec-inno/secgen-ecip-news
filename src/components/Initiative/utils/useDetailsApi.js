import axios from 'axios';
import { useState, useEffect } from 'react';

const useDetailsApi = ({ location, locale }) => {
  const { GATSBY_INITIATIVES_API: api } = process.env;

  const [details, setDetails] = useState({});
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState({});

  useEffect(() => {
    const initiativeId = location.hash.substr(1, location.hash.length);
    const endpoint = `${api}/register/details/${initiativeId}`;

    setIsLoading(true);

    axios
      .get(endpoint)
      .then(response => {
        setDetails(response.data);
        setIsLoading(false);
      })
      .catch(e => {
        setError(e);
        setIsLoading(false);
      });
  }, [locale]);

  return { details, isLoading, error };
};

export default useDetailsApi;
