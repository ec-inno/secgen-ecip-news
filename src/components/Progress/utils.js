export const steps = [
  'REGISTERED',
  'COLLECTION_START_DATE',
  'ONGOING',
  'COLLECTION_EARLY_CLOSURE',
  'CLOSED',
  'VERIFICATION',
  'SUBMITTED',
  'ANSWERED',
];

/**
 * Reorder progress stages to match order of `steps`.
 * @param {Array<Object>} progress
 * @param {String} progress.name
 * @param {Boolean} progress.active
 * @param {Date} progress.date
 */
export const getStages = progress => {
  const stages = [];

  steps.forEach(step => {
    const match = progress.find(item => item.name === step);
    if (match) stages.push(match);
  });

  return stages;
};
