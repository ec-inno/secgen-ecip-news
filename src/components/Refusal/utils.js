import messages from './messages.json';

export const reasonsSupported = [
  'reason.action.registration.reject.abuse',
  'reason.action.registration.reject.competences',
  'reason.action.registration.reject.requirements',
  'reason.action.registration.reject.values',
];

export const getTranslation = ({ code, language }) =>
  messages.find(
    message => message.LNG_CODE === language && message.RTR_KEY === code
  );

export const getReasons = ({ input, language }) => {
  const output = [];

  input.forEach(reason => {
    if (reasonsSupported.includes(reason)) {
      const translation = getTranslation({ code: reason, language });
      if (translation) {
        output.push(translation);
      }
    }
  });

  return output;
};
