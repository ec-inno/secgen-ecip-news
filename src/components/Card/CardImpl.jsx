import React from 'react';
import PropTypes from 'prop-types';

import upperCaseFirstChar from '@eci/utils/upperCaseFirstChar';
import { useLogoApi } from '@eci/utils/useLogoApi';

import Card from './Card';
import defaultImage from '../assets/images/default-image.png';

const CardImpl = ({
  title,
  status,
  logo,
  href,
  totalSupporters,
  supportLink,
}) => {
  if (logo && logo.id && logo.mimeType) {
    const { data } = useLogoApi(logo.id);

    return (
      <Card
        title={title}
        href={href}
        status={upperCaseFirstChar(status)}
        totalSupporters={totalSupporters}
        supportLink={supportLink}
        background={`data:${logo.mimeType};base64,${data}`}
      />
    );
  }

  return (
    <Card
      title={title}
      href={href}
      status={upperCaseFirstChar(status)}
      totalSupporters={totalSupporters}
      supportLink={supportLink}
      background={defaultImage}
    />
  );
};

CardImpl.propTypes = {
  title: PropTypes.string,
  href: PropTypes.string,
  status: PropTypes.string,
  totalSupporters: PropTypes.number,
  supportLink: PropTypes.string,
  logo: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    mimeType: PropTypes.string,
    size: PropTypes.number,
  }),
};

export default CardImpl;
