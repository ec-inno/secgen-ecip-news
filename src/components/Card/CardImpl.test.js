import React from 'react';
import { render } from '@testing-library/react';

import * as useLogoApi from '@eci/utils/useLogoApi';
import CardImpl from './CardImpl';

describe('CardImpl', () => {
  it('render even on missing input parameters', () => {
    const { container } = render(<CardImpl />);
    expect(container).toMatchSnapshot();
  });

  it('renders default image when logo is not provided', () => {
    const props = {
      title: 'Test title',
      href: '#',
      status: 'TEST status',
      totalSupporters: 1000,
      supportLink: '#',
    };

    const { container } = render(<CardImpl {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('displays custom background image when logo is provided', () => {
    jest.spyOn(useLogoApi, 'useLogoApi').mockImplementation(() => '');

    const props = {
      title: 'Test title',
      href: '#',
      status: 'TEST status',
      totalSupporters: 1000,
      supportLink: '#',
      logo: {
        id: 17215,
        name: 'converging.JPG',
        mimeType: 'image/jpeg',
        size: 100869,
      },
    };

    const { container } = render(<CardImpl {...props} />);
    expect(container).toMatchSnapshot();
  });
});
