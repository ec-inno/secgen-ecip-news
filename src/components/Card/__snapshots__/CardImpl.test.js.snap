// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImpl displays custom background image when logo is provided 1`] = `
<div>
  <article
    class="ecl-card"
  >
    <header
      class="ecl-card__header"
    >
      <span
        class="ecl-tag ecl-u-f-r eci-tag"
      >
        Test status
      </span>
      <div
        aria-label="Test title"
        class="ecl-card__image"
        role="img"
        style="background-image: url(data:image/jpeg;base64,undefined);"
      />
      <h1
        class="ecl-card__title"
      >
        <a
          aria-label="Test title"
          class="ecl-link ecl-link--standalone"
          href="#"
          level="1"
        >
          Test title
        </a>
      </h1>
    </header>
    <section
      class="ecl-card__body"
    >
      <p
        class="ecl-u-type-paragraph ecl-u-mv-none"
      >
        1000
         
        supporters
      </p>
      <ul
        class="ecl-card__link-container"
      >
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-card__link ecl-link ecl-link--standalone"
            href="#"
            target="_blank"
          >
            Support this initiative
          </a>
        </li>
      </ul>
    </section>
  </article>
</div>
`;

exports[`CardImpl render even on missing input parameters 1`] = `
<div>
  <article
    class="ecl-card"
  >
    <header
      class="ecl-card__header"
    >
      <span
        class="ecl-tag ecl-u-f-r eci-tag"
      />
      <div
        class="ecl-card__image"
        role="img"
        style="background-image: url(file);"
      />
      <h1
        class="ecl-card__title"
      >
        <a
          class="ecl-link ecl-link--standalone"
          level="1"
        />
      </h1>
    </header>
    <section
      class="ecl-card__body"
    >
      <p
        class="ecl-u-type-paragraph ecl-u-mv-none"
      >
         
        supporters
      </p>
    </section>
  </article>
</div>
`;

exports[`CardImpl renders default image when logo is not provided 1`] = `
<div>
  <article
    class="ecl-card"
  >
    <header
      class="ecl-card__header"
    >
      <span
        class="ecl-tag ecl-u-f-r eci-tag"
      >
        TEST status
      </span>
      <div
        aria-label="Test title"
        class="ecl-card__image"
        role="img"
        style="background-image: url(file);"
      />
      <h1
        class="ecl-card__title"
      >
        <a
          aria-label="Test title"
          class="ecl-link ecl-link--standalone"
          href="#"
          level="1"
        >
          Test title
        </a>
      </h1>
    </header>
    <section
      class="ecl-card__body"
    >
      <p
        class="ecl-u-type-paragraph ecl-u-mv-none"
      >
        1000
         
        supporters
      </p>
      <ul
        class="ecl-card__link-container"
      >
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-card__link ecl-link ecl-link--standalone"
            href="#"
            target="_blank"
          >
            Support this initiative
          </a>
        </li>
      </ul>
    </section>
  </article>
</div>
`;
